### ISSUE TITLE
Issue with interpolate_marcs for metal-poor stars

### ISSUE THREAD

**Issue by segasai** [ajwheeler/Korg.jl#318]
Hi,

Thanks for implementing the interpolate_marcs() code in Korg - that's super useful. 
However I think something is a bit broken with interpolation when going to the metal-poor part of the grid

```julia
julia> X=Korg.interpolate_marcs(5000,1,-2,0.4,0)
Korg.ShellAtmosphere{Float64, Float64, Float64, Float64, Float64, Float64} with 56 layers

julia> X=Korg.interpolate_marcs(5000,1,-3,0.4,0)
ERROR: MethodError: no method matching nextfloat(::Int64)

Closest candidates are:
  nextfloat(::BigFloat)
   @ Base mpfr.jl:1010
  nextfloat(::ForwardDiff.Dual{T, V, N}) where {T, V, N}
   @ ForwardDiff ~/.julia/packages/ForwardDiff/PcZ48/src/dual.jl:309
  nextfloat(::BigFloat, ::Integer)
   @ Base mpfr.jl:1010
  ...

Stacktrace:
  [1] _broadcast_getindex_evalf
    @ ./broadcast.jl:709 [inlined]
  [2] _broadcast_getindex
    @ ./broadcast.jl:682 [inlined]
  [3] getindex
    @ ./broadcast.jl:636 [inlined]
  [4] macro expansion
    @ ./broadcast.jl:1004 [inlined]
  [5] macro expansion
    @ ./simdloop.jl:77 [inlined]
  [6] copyto!
    @ ./broadcast.jl:1003 [inlined]
  [7] copyto!
    @ ./broadcast.jl:956 [inlined]
  [8] materialize!
    @ ./broadcast.jl:914 [inlined]
  [9] materialize!
    @ ./broadcast.jl:911 [inlined]
 [10] lazy_multilinear_interpolation(params::Vector{…}, nodes::Vector{…}, grid::Array{…}; param_names::Vector{…}, perturb_at_grid_values::Bool)
    @ Korg ~/.julia/packages/Korg/0Z2WA/src/lazy_multilinear_interpolation.jl:25
 [11] interpolate_marcs(Teff::Int64, logg::Int64, m_H::Int64, alpha_m::Float64, C_m::Int64; spherical::Bool, perturb_at_grid_values::Bool, resampled_cubic_for_cool_dwarfs::Bool, archives::Tuple{…})
    @ Korg ~/.julia/packages/Korg/0Z2WA/src/atmosphere.jl:301
 [12] interpolate_marcs(Teff::Int64, logg::Int64, m_H::Int64, alpha_m::Float64, C_m::Int64)
    @ Korg ~/.julia/packages/Korg/0Z2WA/src/atmosphere.jl:275
 [13] top-level scope
    @ REPL[5]:1
Some type information was truncated. Use `show(err)` to see complete types.

julia> 

```

I tried to disable the perturb_at_grid_values but it still seem to fail

```julia

julia> X=Korg.interpolate_marcs(5000,1,-3,0.4,0,perturb_at_grid_values=false)
ERROR: InexactError: Int64(3627.3)
Stacktrace:
  [1] Int64
    @ ./float.jl:912 [inlined]
  [2] convert
    @ ./number.jl:7 [inlined]
  [3] setindex!
    @ ./array.jl:1021 [inlined]
  [4] setindex!
    @ ./subarray.jl:343 [inlined]
  [5] copyto_unaliased!(deststyle::IndexLinear, dest::SubArray{…}, srcstyle::IndexLinear, src::SubArray{…})
    @ Base ./abstractarray.jl:1088
  [6] copyto!(dest::SubArray{Int64, 2, Array{…}, Tuple{…}, true}, src::SubArray{Float32, 2, Array{…}, Tuple{…}, true})
    @ Base ./abstractarray.jl:1068
  [7] copyto!
    @ ./broadcast.jl:997 [inlined]
  [8] copyto!
    @ ./broadcast.jl:956 [inlined]
  [9] materialize!
    @ ./broadcast.jl:914 [inlined]
 [10] materialize!(dest::SubArray{…}, bc::Base.Broadcast.Broadcasted{…})
    @ Base.Broadcast ./broadcast.jl:911
 [11] lazy_multilinear_interpolation(params::Vector{…}, nodes::Vector{…}, grid::Array{…}; param_names::Vector{…}, perturb_at_grid_values::Bool)
    @ Korg ~/.julia/packages/Korg/0Z2WA/src/lazy_multilinear_interpolation.jl:55
 [12] interpolate_marcs(Teff::Int64, logg::Int64, m_H::Int64, alpha_m::Float64, C_m::Int64; spherical::Bool, perturb_at_grid_values::Bool, resampled_cubic_for_cool_dwarfs::Bool, archives::Tuple{…})
    @ Korg ~/.julia/packages/Korg/0Z2WA/src/atmosphere.jl:301
 [13] top-level scope
    @ REPL[7]:1
Some type information was truncated. Use `show(err)` to see complete types.
```

Am I doing something wrong here ? 
But if I had to guess the code is getting confused by the fact that there is only one alpha value in that part of the grid.
It would be great to still be able to interpolate over teff,logg,feh in that part of the parameter space.

FWIW (I'm using the latest released Korg version)

Thanks
        S

**Comment by andycasey** [comment]
Hi Sergey,

Do you still get this error if you supply floats for all arguments?

The following seems to work for me (different Korg version, but should be same behaviour), but I do agree that int/floats here should both work.

```
julia> X=Korg.interpolate_marcs(5000,1,-2,0.4,0)
Korg.ShellAtmosphere{Float64, Float64, Float64, Float64, Float64, Float64} with 56 layers

julia> X=Korg.interpolate_marcs(5000.0,1.0,-3.0,0.4,0.0)
Korg.ShellAtmosphere{Float64, Float64, Float64, Float64, Float64, Float64} with 56 layers
```

**Comment by segasai** [comment]
Hi Andy,

Yes it works with floats. Thanks!
I assume the issue is that isexact test somehow fails and alpha is not considered fixed.

**Comment by ajwheeler** [comment]
I have fixed this with #320.  Thanks for reporting!
